### DISCUSSION TITLE
Kafka "end of partition/topic" events

### DISCUSSION THREAD

**Discussion by Gyllsdorff** [spring-projects/spring-kafka#3478] [Q&A]
I have a small log-compacted topic that I need to consume before my application can start to receive HTTP request. When I use the librdkafka based consumer in java, go or Python I can use the [`enable.partition.eof`](https://github.com/confluentinc/librdkafka/blob/master/CONFIGURATION.md#:~:text=enable.partition.eof) config to receive a event every-time I reach the end of a partition which I use to keep keep track if I hit the high watermark at least once for each partition of my topic.

Is it possible to something similar in Spring Kafka? The closest things I could find in the documentation is the "[Detecting Idle and Non-Responsive Consumers](https://docs.spring.io/spring-kafka/reference/kafka/events.html#idle-containers)" but that relies on hardcoded idle timeouts.

**Comment by sobychacko** [comment]
@Gyllsdorff Does `ListenerContainerPartitionIdleEvent` work for that use case? For, e.g. you can listen for that event, and as soon as the partition goes idle after the configured idle interval, you can assume that the partition is caught up. Firing an event each time we get caught up with the end of the partition might introduce a scenario in which you could receive several such events.
